src/Core/src/Platform/iOS/UIPageControlExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Maui.Graphics;
+using UIKit;
+
+namespace Microsoft.Maui.Platform.iOS
+{
+	public static class UIPageControlExtensions
+	{
+		public static void UpdateIndicatorShape(this MauiPageControl pageControl, IIndicatorView indicatorView)
+		{
+			pageControl.IsSquare = !IsCircleShape(indicatorView);
+
+			bool IsCircleShape(IIndicatorView indicatorView)
```

Comment:
Maybe this can also be an internal extension as it is used in several places.

Reply:
done